W0427 17:30:11.759493 2847463 torch/distributed/run.py:793] 
W0427 17:30:11.759493 2847463 torch/distributed/run.py:793] *****************************************
W0427 17:30:11.759493 2847463 torch/distributed/run.py:793] Setting OMP_NUM_THREADS environment variable for each process to be 1 in default, to avoid your system being overloaded, please further tune the variable for optimal performance in your application as needed. 
W0427 17:30:11.759493 2847463 torch/distributed/run.py:793] *****************************************
Data loading | DONE!
Data loading | DONE!
2025-04-27 17:30:28,288 INFO     -------------------------
2025-04-27 17:30:28,288 INFO     | Pre-train Generator
2025-04-27 17:30:28,288 INFO     -------------------------
mode:  warmup
mode:  warmup
2025-04-27 17:30:28,435 INFO     >>>>> Generator: Training
2025-04-27 17:30:34,323 INFO     [E1000] Total: 2.549636 | Main: 2.625639 | Aux: 1.083557
2025-04-27 17:30:39,385 INFO     [E2000] Total: 2.468215 | Main: 2.542344 | Aux: 1.059760
2025-04-27 17:30:44,751 INFO     [E3000] Total: 2.454475 | Main: 2.527996 | Aux: 1.057586
2025-04-27 17:30:49,838 INFO     [E4000] Total: 2.445863 | Main: 2.519064 | Aux: 1.055031
2025-04-27 17:30:55,298 INFO     [E5000] Total: 2.438873 | Main: 2.511730 | Aux: 1.054599
2025-04-27 17:31:00,366 INFO     [E6000] Total: 2.435323 | Main: 2.508057 | Aux: 1.053368
2025-04-27 17:31:05,433 INFO     [E7000] Total: 2.432158 | Main: 2.504753 | Aux: 1.052856
2025-04-27 17:31:10,835 INFO     [E8000] Total: 2.427399 | Main: 2.499773 | Aux: 1.052296
2025-04-27 17:31:15,908 INFO     [E9000] Total: 2.419945 | Main: 2.491948 | Aux: 1.051898
2025-04-27 17:31:21,279 INFO     [E10000] Total: 2.421701 | Main: 2.493814 | Aux: 1.051568
2025-04-27 17:31:21,283 INFO     -------------------------
2025-04-27 17:31:21,283 INFO     | EM Iteration: 1/1
2025-04-27 17:31:21,283 INFO     -------------------------
2025-04-27 17:31:21,283 INFO     >>>>> Generator: Rule generation with sampling
2025-04-27 17:31:21,488 INFO     Predictor: read 500 rules from list.
2025-04-27 17:31:21,488 INFO     Initializing distributed process group
2025-04-27 17:31:21,501 INFO     Predictor: read 500 rules from list.
2025-04-27 17:31:21,503 INFO     Preprocess training set
2025-04-27 17:31:21,577 INFO     >>>>> Predictor: Training
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.9/logging/__init__.py", line 1083, in emit
    msg = self.format(record)
  File "/usr/lib/python3.9/logging/__init__.py", line 927, in format
    return fmt.format(record)
  File "/usr/lib/python3.9/logging/__init__.py", line 663, in format
    record.message = record.getMessage()
  File "/usr/lib/python3.9/logging/__init__.py", line 367, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "/home/emma/emma/bilab_archive/polly/RNNLogic/src/run_rnnlogic.py", line 181, in <module>
    main(parse_args())
  File "/home/emma/emma/bilab_archive/polly/RNNLogic/src/run_rnnlogic.py", line 106, in main
    solver_p.train(**cfg.predictor.train)
  File "/home/emma/emma/bilab_archive/polly/RNNLogic/src/trainer.py", line 86, in train
    logits, mask = model(all_h, all_r, edges_to_remove)
  File "/home/emma/polly_env_py39/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/emma/polly_env_py39/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/emma/polly_env_py39/lib/python3.9/site-packages/torch/nn/parallel/distributed.py", line 1643, in forward
    else self._run_ddp_forward(*inputs, **kwargs)
  File "/home/emma/polly_env_py39/lib/python3.9/site-packages/torch/nn/parallel/distributed.py", line 1459, in _run_ddp_forward
    return self.module(*inputs, **kwargs)  # type: ignore[index]
  File "/home/emma/polly_env_py39/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/emma/polly_env_py39/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/emma/emma/bilab_archive/polly/RNNLogic/src/predictors.py", line 63, in forward
    x = self.graph.grounding(all_h, r_head, r_body, edges_to_remove)
  File "/home/emma/emma/bilab_archive/polly/RNNLogic/src/data.py", line 156, in grounding
    x = self.propagate(x, r_body)
  File "/home/emma/emma/bilab_archive/polly/RNNLogic/src/data.py", line 185, in propagate
    logging.info("message size: ", message.size())
Message: 'message size: '
Arguments: (torch.Size([6102, 1]),)
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.9/logging/__init__.py", line 1083, in emit
    msg = self.format(record)
  File "/usr/lib/python3.9/logging/__init__.py", line 927, in format
    return fmt.format(record)
  File "/usr/lib/python3.9/logging/__init__.py", line 663, in format
    record.message = record.getMessage()
  File "/usr/lib/python3.9/logging/__init__.py", line 367, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "/home/emma/emma/bilab_archive/polly/RNNLogic/src/run_rnnlogic.py", line 181, in <module>
    main(parse_args())
  File "/home/emma/emma/bilab_archive/polly/RNNLogic/src/run_rnnlogic.py", line 106, in main
    solver_p.train(**cfg.predictor.train)
  File "/home/emma/emma/bilab_archive/polly/RNNLogic/src/trainer.py", line 86, in train
    logits, mask = model(all_h, all_r, edges_to_remove)
  File "/home/emma/polly_env_py39/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/emma/polly_env_py39/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/emma/polly_env_py39/lib/python3.9/site-packages/torch/nn/parallel/distributed.py", line 1643, in forward
    else self._run_ddp_forward(*inputs, **kwargs)
  File "/home/emma/polly_env_py39/lib/python3.9/site-packages/torch/nn/parallel/distributed.py", line 1459, in _run_ddp_forward
    return self.module(*inputs, **kwargs)  # type: ignore[index]
  File "/home/emma/polly_env_py39/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/emma/polly_env_py39/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/emma/emma/bilab_archive/polly/RNNLogic/src/predictors.py", line 63, in forward
    x = self.graph.grounding(all_h, r_head, r_body, edges_to_remove)
  File "/home/emma/emma/bilab_archive/polly/RNNLogic/src/data.py", line 156, in grounding
    x = self.propagate(x, r_body)
  File "/home/emma/emma/bilab_archive/polly/RNNLogic/src/data.py", line 185, in propagate
    logging.info("message size: ", message.size())
Message: 'message size: '
Arguments: (torch.Size([6102, 1]),)
[rank1]: Traceback (most recent call last):
[rank1]:   File "/home/emma/emma/bilab_archive/polly/RNNLogic/src/run_rnnlogic.py", line 181, in <module>
[rank1]:     main(parse_args())
[rank1]:   File "/home/emma/emma/bilab_archive/polly/RNNLogic/src/run_rnnlogic.py", line 106, in main
[rank1]:     solver_p.train(**cfg.predictor.train)
[rank1]:   File "/home/emma/emma/bilab_archive/polly/RNNLogic/src/trainer.py", line 86, in train
[rank1]:     logits, mask = model(all_h, all_r, edges_to_remove)
[rank1]:   File "/home/emma/polly_env_py39/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
[rank1]:     return self._call_impl(*args, **kwargs)
[rank1]:   File "/home/emma/polly_env_py39/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
[rank1]:     return forward_call(*args, **kwargs)
[rank1]:   File "/home/emma/polly_env_py39/lib/python3.9/site-packages/torch/nn/parallel/distributed.py", line 1643, in forward
[rank1]:     else self._run_ddp_forward(*inputs, **kwargs)
[rank1]:   File "/home/emma/polly_env_py39/lib/python3.9/site-packages/torch/nn/parallel/distributed.py", line 1459, in _run_ddp_forward
[rank1]:     return self.module(*inputs, **kwargs)  # type: ignore[index]
[rank1]:   File "/home/emma/polly_env_py39/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
[rank1]:     return self._call_impl(*args, **kwargs)
[rank1]:   File "/home/emma/polly_env_py39/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
[rank1]:     return forward_call(*args, **kwargs)
[rank1]:   File "/home/emma/emma/bilab_archive/polly/RNNLogic/src/predictors.py", line 63, in forward
[rank1]:     x = self.graph.grounding(all_h, r_head, r_body, edges_to_remove)
[rank1]:   File "/home/emma/emma/bilab_archive/polly/RNNLogic/src/data.py", line 156, in grounding
[rank1]:     x = self.propagate(x, r_body)
[rank1]:   File "/home/emma/emma/bilab_archive/polly/RNNLogic/src/data.py", line 195, in propagate
[rank1]:     E, B, D = message.size()
[rank1]: ValueError: not enough values to unpack (expected 3, got 2)
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.9/logging/__init__.py", line 1083, in emit
    msg = self.format(record)
  File "/usr/lib/python3.9/logging/__init__.py", line 927, in format
    return fmt.format(record)
  File "/usr/lib/python3.9/logging/__init__.py", line 663, in format
    record.message = record.getMessage()
  File "/usr/lib/python3.9/logging/__init__.py", line 367, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "/home/emma/emma/bilab_archive/polly/RNNLogic/src/run_rnnlogic.py", line 181, in <module>
    main(parse_args())
  File "/home/emma/emma/bilab_archive/polly/RNNLogic/src/run_rnnlogic.py", line 106, in main
    solver_p.train(**cfg.predictor.train)
  File "/home/emma/emma/bilab_archive/polly/RNNLogic/src/trainer.py", line 86, in train
    logits, mask = model(all_h, all_r, edges_to_remove)
  File "/home/emma/polly_env_py39/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/emma/polly_env_py39/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/emma/polly_env_py39/lib/python3.9/site-packages/torch/nn/parallel/distributed.py", line 1643, in forward
    else self._run_ddp_forward(*inputs, **kwargs)
  File "/home/emma/polly_env_py39/lib/python3.9/site-packages/torch/nn/parallel/distributed.py", line 1459, in _run_ddp_forward
    return self.module(*inputs, **kwargs)  # type: ignore[index]
  File "/home/emma/polly_env_py39/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/emma/polly_env_py39/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/emma/emma/bilab_archive/polly/RNNLogic/src/predictors.py", line 63, in forward
    x = self.graph.grounding(all_h, r_head, r_body, edges_to_remove)
  File "/home/emma/emma/bilab_archive/polly/RNNLogic/src/data.py", line 156, in grounding
    x = self.propagate(x, r_body)
  File "/home/emma/emma/bilab_archive/polly/RNNLogic/src/data.py", line 185, in propagate
    logging.info("message size: ", message.size())
Message: 'message size: '
Arguments: (torch.Size([11419, 1]),)
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.9/logging/__init__.py", line 1083, in emit
    msg = self.format(record)
  File "/usr/lib/python3.9/logging/__init__.py", line 927, in format
    return fmt.format(record)
  File "/usr/lib/python3.9/logging/__init__.py", line 663, in format
    record.message = record.getMessage()
  File "/usr/lib/python3.9/logging/__init__.py", line 367, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "/home/emma/emma/bilab_archive/polly/RNNLogic/src/run_rnnlogic.py", line 181, in <module>
    main(parse_args())
  File "/home/emma/emma/bilab_archive/polly/RNNLogic/src/run_rnnlogic.py", line 106, in main
    solver_p.train(**cfg.predictor.train)
  File "/home/emma/emma/bilab_archive/polly/RNNLogic/src/trainer.py", line 86, in train
    logits, mask = model(all_h, all_r, edges_to_remove)
  File "/home/emma/polly_env_py39/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/emma/polly_env_py39/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/emma/polly_env_py39/lib/python3.9/site-packages/torch/nn/parallel/distributed.py", line 1643, in forward
    else self._run_ddp_forward(*inputs, **kwargs)
  File "/home/emma/polly_env_py39/lib/python3.9/site-packages/torch/nn/parallel/distributed.py", line 1459, in _run_ddp_forward
    return self.module(*inputs, **kwargs)  # type: ignore[index]
  File "/home/emma/polly_env_py39/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/emma/polly_env_py39/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/emma/emma/bilab_archive/polly/RNNLogic/src/predictors.py", line 63, in forward
    x = self.graph.grounding(all_h, r_head, r_body, edges_to_remove)
  File "/home/emma/emma/bilab_archive/polly/RNNLogic/src/data.py", line 156, in grounding
    x = self.propagate(x, r_body)
  File "/home/emma/emma/bilab_archive/polly/RNNLogic/src/data.py", line 185, in propagate
    logging.info("message size: ", message.size())
Message: 'message size: '
Arguments: (torch.Size([11419, 1]),)
[rank0]: Traceback (most recent call last):
[rank0]:   File "/home/emma/emma/bilab_archive/polly/RNNLogic/src/run_rnnlogic.py", line 181, in <module>
[rank0]:     main(parse_args())
[rank0]:   File "/home/emma/emma/bilab_archive/polly/RNNLogic/src/run_rnnlogic.py", line 106, in main
[rank0]:     solver_p.train(**cfg.predictor.train)
[rank0]:   File "/home/emma/emma/bilab_archive/polly/RNNLogic/src/trainer.py", line 86, in train
[rank0]:     logits, mask = model(all_h, all_r, edges_to_remove)
[rank0]:   File "/home/emma/polly_env_py39/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
[rank0]:     return self._call_impl(*args, **kwargs)
[rank0]:   File "/home/emma/polly_env_py39/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
[rank0]:     return forward_call(*args, **kwargs)
[rank0]:   File "/home/emma/polly_env_py39/lib/python3.9/site-packages/torch/nn/parallel/distributed.py", line 1643, in forward
[rank0]:     else self._run_ddp_forward(*inputs, **kwargs)
[rank0]:   File "/home/emma/polly_env_py39/lib/python3.9/site-packages/torch/nn/parallel/distributed.py", line 1459, in _run_ddp_forward
[rank0]:     return self.module(*inputs, **kwargs)  # type: ignore[index]
[rank0]:   File "/home/emma/polly_env_py39/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
[rank0]:     return self._call_impl(*args, **kwargs)
[rank0]:   File "/home/emma/polly_env_py39/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
[rank0]:     return forward_call(*args, **kwargs)
[rank0]:   File "/home/emma/emma/bilab_archive/polly/RNNLogic/src/predictors.py", line 63, in forward
[rank0]:     x = self.graph.grounding(all_h, r_head, r_body, edges_to_remove)
[rank0]:   File "/home/emma/emma/bilab_archive/polly/RNNLogic/src/data.py", line 156, in grounding
[rank0]:     x = self.propagate(x, r_body)
[rank0]:   File "/home/emma/emma/bilab_archive/polly/RNNLogic/src/data.py", line 195, in propagate
[rank0]:     E, B, D = message.size()
[rank0]: ValueError: not enough values to unpack (expected 3, got 2)
[rank0]:[W427 17:31:23.760298803 ProcessGroupNCCL.cpp:1250] Warning: WARNING: process group has NOT been destroyed before we destruct ProcessGroupNCCL. On normal program exit, the application should call destroy_process_group to ensure that any pending NCCL operations have finished in this process. In rare cases this process can exit before this point and block the progress of another member of the process group. This constraint has always been present,  but this warning has only been added since PyTorch 2.4 (function operator())
E0427 17:31:24.028703 2847463 torch/distributed/elastic/multiprocessing/api.py:869] failed (exitcode: 1) local_rank: 0 (pid: 2847481) of binary: /home/emma/polly_env_py39/bin/python3.9
Traceback (most recent call last):
  File "/home/emma/polly_env_py39/bin/torchrun", line 8, in <module>
    sys.exit(main())
  File "/home/emma/polly_env_py39/lib/python3.9/site-packages/torch/distributed/elastic/multiprocessing/errors/__init__.py", line 355, in wrapper
    return f(*args, **kwargs)
  File "/home/emma/polly_env_py39/lib/python3.9/site-packages/torch/distributed/run.py", line 919, in main
    run(args)
  File "/home/emma/polly_env_py39/lib/python3.9/site-packages/torch/distributed/run.py", line 910, in run
    elastic_launch(
  File "/home/emma/polly_env_py39/lib/python3.9/site-packages/torch/distributed/launcher/api.py", line 138, in __call__
    return launch_agent(self._config, self._entrypoint, list(args))
  File "/home/emma/polly_env_py39/lib/python3.9/site-packages/torch/distributed/launcher/api.py", line 269, in launch_agent
    raise ChildFailedError(
torch.distributed.elastic.multiprocessing.errors.ChildFailedError: 
============================================================
run_rnnlogic.py FAILED
------------------------------------------------------------
Failures:
[1]:
  time      : 2025-04-27_17:31:24
  host      : emma-System-Product-Name
  rank      : 1 (local_rank: 1)
  exitcode  : 1 (pid: 2847482)
  error_file: <N/A>
  traceback : To enable traceback see: https://pytorch.org/docs/stable/elastic/errors.html
------------------------------------------------------------
Root Cause (first observed failure):
[0]:
  time      : 2025-04-27_17:31:24
  host      : emma-System-Product-Name
  rank      : 0 (local_rank: 0)
  exitcode  : 1 (pid: 2847481)
  error_file: <N/A>
  traceback : To enable traceback see: https://pytorch.org/docs/stable/elastic/errors.html
============================================================
